### ISSUE TITLE
New buzzwords Could an AI Generate the First Line of a Novel?

### ISSUE THREAD

**Issue by postmillenial** [postmillenial/reading-list#1899]
Could an AI Generate the First Line of a Novel?<br>
http://ift.tt/2iFgSPU<br>
<img width="640" height="426" src="http://ift.tt/2AHd1sE" class="webfeedsFeaturedVisual wp-post-image wp-stateless-item" style="display: block; margin: auto; margin-bottom: 20px;" data-image-size="large" data-stateless-media-bucket="cdn.thenewstack.io" data-stateless-media-name="media/2017/12/f34916b9-books-1245690_640.jpg">
<p>Are we approaching a day where machines can out-perform us in &ldquo;human&rdquo; fields like the creative arts? If you gave the proper training set to a sophisticated neural network, would it eventually be able to compose the first line of a novel?</p>
<p>Not quite yet, as it turns out.</p>
<p>Janelle Shane has already started exploring other questions like this, by fooling around with a neural network. Shane&rsquo;s methodology is simple. &ldquo;All I have to do is give the neural network a long list of examples and it will try its best to teach itself to generate more like them,&rdquo; she <a href="http://ift.tt/2zeNRk7" class="ext-link" rel="external nofollow">explains in one post</a>. And behind all the tomfoolery there&rsquo;s some high-powered technology, she says in an <a href="http://ift.tt/2xR8sY5" class="ext-link" rel="external nofollow">interview with The Next Web</a>.</p>
<p><em>I&rsquo;m using an open-source neural network framework called <a href="http://ift.tt/1Ao3IJ8" class="ext-link" rel="external nofollow">char-rnn</a>, written by Andrej Karpathy in <a href="http://torch.ch/" class="ext-link" rel="external nofollow">Torch</a>. There are all sorts of fancy approaches people use to make neural networks better at understanding and generating sentences and paragraphs. I&rsquo;m using none of that, just a machine learning algorithm that learns to spell words letter by letter with no sense of what they mean.</em></p>
<p>TheNextWeb concluded, &ldquo;So perhaps it isn&rsquo;t the singularity.&rdquo;</p>
<p>But part of the fun is picking something that seems uniquely human. &ldquo;Usually in my experiment, I give the neural network an unfair dataset &mdash; like paint colors,&rdquo; <a href="http://ift.tt/2je29eG" class="ext-link" rel="external nofollow">Shane writes</a>, &ldquo;and it tries its best, but ends up with something unintentionally weird, like a brownish color called Stanky Bean, or a bright blue color called Dad.&rdquo;</p>
<p>And of course, the other half of the fun is sharing the ridiculous results <a href="http://aiweirdness.com/" class="ext-link" rel="external nofollow">on her blog</a>. For example, given a dataset of 37,265 fish names, the neural network tried its hardest to <a href="http://ift.tt/2je29eG" class="ext-link" rel="external nofollow">generate its own</a>. And last week Shane trained it <a href="http://ift.tt/2AV0C1A" class="ext-link" rel="external nofollow">on the first lines from 224 Disney songs</a>. But this month Shane turned her attention to one of the most human endeavors of all: writing a novel. &ldquo;When you&rsquo;re faced with a blank page, sometimes it&rsquo;s just hard to get started,&rdquo; Shane <a href="http://ift.tt/2zeNRk7" class="ext-link" rel="external nofollow">joked on her web page</a>. &ldquo;I wanted to see if I could train a computer program to help.&rdquo;</p>
<p>&ldquo;Turns out, she couldn&rsquo;t,&rdquo; <a href="http://ift.tt/2lMpEwt" class="ext-link" rel="external nofollow">quipped Geek.com</a>.</p>
<p>For starters, it was hard to find enough data for training the neural network. After searching for first lines from novels, Shane only found a few hundred, where &ldquo;ideally, I need thousands.&rdquo; Otherwise, the neural network suggests novel first lines which are nearly identical to its training data, or clumsily pastes two together, with a few oddball tweaks. At one point it suggested a novel that started with Snoopy&rsquo;s favorite opening clause but ended with the first line of James Joyce&rsquo;s &ldquo;Ulysses,&rdquo; adding in a few random words for good measure.</p>
<p><em><tt>"It was a dark and stormy night; the rain fell in torrents &mdash; except the station steps; plump Buck Mulligan came from the stairhead, bearing a bowl of people."</tt></em></p>
<p>In &ldquo;Ulyssess&rdquo; Buck Mulligan bears &ldquo;a bowl of lather on which a mirror and a razor lay crossed.&rdquo; (Although admittedly it&rsquo;s much more interesting if the bowl is filled with <em>people</em>). But Shane wasn&rsquo;t satisfied with the novel openings generated by the neural network. &ldquo;Most didn&rsquo;t make much sense, and/or were obvious mishmashes of famous lines.&rdquo;</p>
<p>Though a few of them were oddly poignant.</p>
<p><em><tt>"It was a wrong number that struggled against the darkness."</tt></em></p>
<p><em><tt>"The sky above the present century had reached the snapping point."</tt></em></p>
<p><em><tt>"The snow is gone sometime, and you said, Why, and I said, To be with the darkness."</tt></em></p>
<p>Weeks later Shane admitted that &ldquo;It didn&rsquo;t go so well.&rdquo; She&rsquo;d found one more site offering an additional 900 first sentences for novels:&nbsp;The Bulwer-Lytton Fiction Contest challenges competitors to compose the <em>worst</em> possible first line for a novel. (For <a href="http://ift.tt/1ux8TPG" class="ext-link" rel="external nofollow">example</a>, &ldquo;This is a tale of love, pain, loss, and redemption &mdash; and of a baboon, Amelia.&rdquo;)</p>
<p>Shane&rsquo;s verdict? &ldquo;It didn&rsquo;t help.&rdquo; She ended up with novel openings like&hellip;.</p>
<p><em><tt>"It was a dark and stormy night and the secret being a silver-backed gorilla."</tt></em></p>
<p>In desperation, she turned to the internet &mdash; or at least, to the community of her site&rsquo;s readers and her followers on Twitter. Yes, she asked strangers on the internet to send in the first lines from <em>their</em> favorite novels &mdash; or even from amateur novels they&rsquo;d written themselves. Could her neural network craft a convincing opening for a novel from a crowdsourced dataset contributed by human volunteers?</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">awesome, thanks so much! Up to 175 entries already.</p>
<p>&mdash; Janelle Shane (@JanelleCShane) <a href="https://twitter.com/JanelleCShane/status/926099341391028226?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 2, 2017</a></p>
</blockquote>
</div>
<p>Or would online humans simply overwhelm it with their anomalous affinity for science fiction?</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">too late. Already got 3 Neuromancers. I can filter out duplicates later though</p>
<p>&mdash; Janelle Shane (@JanelleCShane) <a href="https://twitter.com/JanelleCShane/status/926101744001544195?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 2, 2017</a></p>
</blockquote>
</div>
<p>Shane waited patiently until the end of the month, even spending the week before Thanksgiving on an entirely different experiment &mdash; training her AI network to <a href="http://ift.tt/2jJ3qLc" class="ext-link" rel="external nofollow">generate names of pies</a>, based on the names of 2,237 pie recipes.) But soon it was back to AI-generated novels,&nbsp;or at least, collecting enough data to give her neural network a sufficient training set.</p>
<p>Fortunately, at this stage, she got a little help from her friends:</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Help <a href="https://twitter.com/JanelleCShane?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">@JanelleCShane</a>'s pet neural network write a novel, bc even artificial intelligence suffers from writer's block. <a href="https://t.co/7Rz91nSp7C" class="ext-link" rel="external nofollow">https://t.co/7Rz91nSp7C</a></p>
<p>&mdash; Electric Literature (@ElectricLit) <a href="https://twitter.com/ElectricLit/status/926463997649571841?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 3, 2017</a></p>
</blockquote>
</div>
<p>And soon, helpful submissions started pouring in&hellip;.</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Ha, someone did just submit a bunch of kid's books.<br>
9661 entries now! <a href="https://t.co/Gf1vqlDRf7" class="ext-link" rel="external nofollow">http://pic.twitter.com/Gf1vqlDRf7</a></p>
<p>&mdash; Janelle Shane (@JanelleCShane) <a href="https://twitter.com/JanelleCShane/status/933375702044368897?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 22, 2017</a></p>
</blockquote>
</div>
<p>&nbsp;</p>
<p>Meanwhile, another throng was gathering on a different part of the internet. On Thursday, at the stroke of midnight, 384,126 aspiring novel-writers marked the end of &ldquo;National Novel Writing Month&rdquo; &mdash; the 18th annual attempt to crank out a quickie novel in 30 days using nothing but their human wits. Shane chose that same momentous day to finally <a href="http://ift.tt/2j3npkq" class="ext-link" rel="external nofollow">reveal the results of her own experiment</a>. Her readers had come through with an additional 11,135 additional first lines from novels.</p>
<p>&ldquo;folks, you have made me and the neural network so very happy.&rdquo;</p>
<p>Combining the user-submitted first lines with the 900 that she already had, Shane created a robust training dataset with 10,096 first lines of novels. Sounding more upbeat, she wrote that &ldquo;The first results showed, if not promise, then at least evidence of the high number of &ldquo;<tt>My Little Pony"</tt> stories in the dataset.&rdquo;</p>
<p><em><tt>"...an of the the all stood ponyville at es that ev the."</tt></em></p>
<p>But first results are always a little rough, and Shane reported that after training longer the neural network &ldquo;soon made some improvement, and once in a while would produce a grammatically correct sentence as long as it was very short.&rdquo;</p>
<p><em><tt>"It was an hour of the night."</tt></em></p>
<p><em><tt>"I know they are from the mountain."</tt></em></p>
<p>And it became eerily more human after a few more iterations through its training set. Or at least, as Shane writes, &ldquo;It learned eventually how to begin a book by talking about the weather &mdash; although not always successfully.&rdquo;</p>
<p><em><tt>"The night is like a wounded carpets from the Crumzon."</tt></em></p>
<p><em><tt>"The sky was dead."</tt></em></p>
<p>The training continued and continued, and eventually, Shane reports, many of the lines &ldquo;almost made sense&hellip;especially the shortest ones.&rdquo;</p>
<p><em><tt>"I am not a king."</tt></em></p>
<p><em><tt>"The sky has gone."</tt></em></p>
<p><em><tt>"The sun was coming."</tt></em></p>
<p><em><tt>"The night was over."</tt></em></p>
<p>By the end of the experiment, she&rsquo;d concluded that some of the AI-generated first lines &ldquo;were actually rather intriguing. I might read these books.&rdquo;</p>
<p><em><tt>"The silence was unlike a place."</tt></em></p>
<p><em><tt>&ldquo;I am forced to write to my neighbors about the beast.&rdquo;</tt></em></p>
<p><em><tt>"The sun was probably for his wife."</tt></em></p>
<p><em><tt>"I am a story that was not a truth."</tt></em></p>
<p>But along the way, she&rsquo;d made an interesting discovery: she wasn&rsquo;t the only one who was trying to use technology to write fiction. For the last few years national novel-writing month has been accompanied by an overlapping event: &ldquo;<a href="http://ift.tt/2AIpsV2" class="local-link">National Novel Generating Month</a>.&rdquo; A community of geeks has been sharing their computer-generated novels <a href="http://ift.tt/2xFvLby" class="ext-link" rel="external nofollow">on a GitHub repository</a>, and Shane couldn&rsquo;t resist joining in the fun.</p>
<p>So, in the end, her post about computer-generated lines for novels ended with a plot twist of its own. &ldquo;This project is <em>my</em> entry for National Novel Generating Month, which means that I generated 140,000 words&rsquo; worth of first lines&hellip;&rdquo;</p>
<p>And surprisingly, some parts of it are actually readable.</p>
<p><em><tt>On the end of April, that was a large little boy before the mast. The book was nothing but stars, and the earth was hard to breathe. The day was enveloped in the year 1874.</tt></em></p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Stephen King clearly wrote this. And he was on potent psychedelics when he did so.</p>
<p>&mdash; Phil Atkin (@RamonesKaraoke) <a href="https://twitter.com/RamonesKaraoke/status/936324748774174721?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 30, 2017</a></p>
</blockquote>
</div>
<p>Shane (and her community of followers) could now finally witness the results of their unholy collaboration.</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Oooh you must be getting these from the big dataset! I haven't looked through it yet. These are nice gems &ndash; share more if you find them!</p>
<p>&mdash; Janelle Shane (@JanelleCShane) <a href="https://twitter.com/JanelleCShane/status/936341014121345024?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 30, 2017</a></p>
</blockquote>
</div>
<p>So what happens when you combine machine learning technology, crowdsourcing, and a human sense of humor?</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">Can't argue with this:<br>
The first day of the world was a mistake.</p>
<p>&mdash; Dr Headgear (@DrHeadgear) <a href="https://twitter.com/DrHeadgear/status/936344871622803456?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 30, 2017</a></p>
</blockquote>
</div>
<p>When challenged to write what it could never fully understand, the neural network still came back with some delightful surprises.</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">I am a decent man who had been stranded to be in the last seat.</p>
<p>&mdash; Dr Headgear (@DrHeadgear) <a href="https://twitter.com/DrHeadgear/status/936352070394892288?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 30, 2017</a></p>
</blockquote>
</div>
<p>Of course, there were a few glitches, but they were apparently caused by human pranksters.</p>
<div class="twitter-tweet">
<blockquote class="twitter-tweet" data-lang="en">
<p lang="en" dir="ltr" xml:lang="en">So sand gets everywhere? Was it coarse and irritating too?</p>
<p>&mdash; Practical Tim (@PracticalPeng) <a href="https://twitter.com/PracticalPeng/status/936326567751503872?ref_src=twsrc%5Etfw" class="ext-link" rel="external nofollow">November 30, 2017</a></p>
</blockquote>
</div>
<p>The experiment appeared to be a success, even if it didn&rsquo;t lead humanity to the perfect machine-generated novel.</p>
<p>Because at least it gave us poor little ponies, lost in the darkness, that most human gift of all.</p>
<p>Laughter.</p>
<hr>
<h1>WebReduce</h1>
<ul>
<li>Amazon announces <a href="http://ift.tt/2AIe7EY" class="ext-link" rel="external nofollow">real-time language translation for text</a>.</li>
<li>How someone turned a Furby <a href="http://ift.tt/2BeE7Ef" class="ext-link" rel="external nofollow">into an Amazon Echo</a>.</li>
<li>Dozens of drones are <a href="http://ift.tt/2zMCbX7" class="ext-link" rel="external nofollow">making deliveries in Switzerland</a>.</li>
<li>&ldquo;FakeSpot&rdquo; tries to stop <a href="http://ift.tt/2BlFUs7" class="ext-link" rel="external nofollow">cheating businesses online</a> with an <a href="http://ift.tt/2fBnkTx" class="ext-link" rel="external nofollow">automated analysis of their Amazon reviews</a>.</li>
<li>Charting &ldquo;<a href="http://ift.tt/2iaCBvO" class="ext-link" rel="external nofollow">the most dramatic rises and falls&rdquo; in technology popularity</a> (based on tag-frequency in Stack Overflow questions).</li>
<li>Study finds it&rsquo;s <a href="http://ift.tt/2zWWBMB" class="ext-link" rel="external nofollow">bad to use laptops in meetings</a>.</li>
<li>How many young Americans are <a href="http://ift.tt/2jhvyRE" class="ext-link" rel="external nofollow">leaving their desk jobs to farm</a>.</li>
</ul>
<p class="attribution">Feature image <a href="http://ift.tt/2gXx0rq" class="ext-link" rel="external nofollow">via</a> Pixabay.</p>
<p>The post <a rel="nofollow" href="http://ift.tt/2iFgSPU">Could an AI Generate the First Line of a Novel?</a> appeared first on <a rel="nofollow" href="https://thenewstack.io">The New Stack</a>.</p>
<br>
<br>
via Blog &ndash; The New Stack https://thenewstack.io<br>
December 03, 2017 at 09:00AM
